addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -455,12 +455,25 @@ function factory(dependencies) {
                 channel.members.includes(this.env.messaging.currentPartner)
             );
 
-            channel = this.env.models['mail.thread'].insert(Object.assign(
-                { isPinned: true },
-                convertedData,
-            ));
+            if (channel) {
+                if (channel.pinnedStatus === 'unpinned' && convertedData.serverFoldState === 'closed') {
+                    channel.delete();
+                    channel = null;
+                } else {
```

Comment:
Don't bother deleting channel.
It makes code much more complex, and this only applies in this flow.
Also I think this is wrong and may introduce subtle bugs, because you think it's safe to delete thread when you consider its integration with chat windows, but you completely omit other (present and future) aspects that **require** keeping knowledge of thread!